apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/CommonQueries.java

```diff
@@ -0,0 +1,211 @@
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import gov.cms.bfd.model.rif.entities.Beneficiary;
+import gov.cms.bfd.model.rif.entities.Beneficiary_;
+import gov.cms.bfd.server.sharedutils.BfdMDC;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.JoinType;
+import javax.persistence.criteria.Root;
+
+/** Common database queries that can be shared between V1 and V2. */
+public class CommonQueries {
+
+  /**
+   * Database function that checks all claims for a given beneficiaryId and returns a bitwise mask
+   * value that shows if a given claim type will have any data.
+   */
+  public static final String CHECK_CLAIMS_FOR_DATA_SQL =
+      "SELECT * FROM check_claims_mask(:beneIdValue)";
+
+  /**
+   * Database function that takes in a patient identifier type and value and returns a bigint
+   * BENE_ID if the identifier can find a beneficiaries record. Supported identifier types are: mbi,
+   * mbi-hash, hicn-hash.
+   */
+  public static final String FIND_BENE_ID_FROM_IDENTIFIER_SQL =
+      "SELECT * FROM find_beneficiary(:searchIdType, :searchIdValue)";
+
+  /** preclude construction from outsiders. */
+  private CommonQueries() {}
+
+  /**
+   * Query database to determine which claim types have data for the specified beneficiary. The
+   * database function returns a bitwise mask value that denotes which claim types will have data
+   * for the specified beneficiaryId. This represents a fast and efficient way to ignore a requested
+   * claim type that ultimately has no data for our beneficiaryId.
+   *
+   * <p>The database function returns bits as follows:
+   *
+   * <ul>
+   *   <li>CARRIER CLAIMS : bit 0
+   *   <li>INPATIENT CLAIMS : bit 1
+   *   <li>OUTPATIENT CLAIMS : bit 2
+   *   <li>SNF CLAIMS : bit 3
+   *   <li>DME CLAIMS : bit 4
+   *   <li>HHA CLAIMS : bit 5
+   *   <li>HOSPICE CLAIMS : bit 6
+   *   <li>PART D CLAIMS : bit 7
+   * </ul>
+   *
+   * <p>For more information on the database function, see:
+   * V111__SETUP_CLAIMS_AVAILABILITY_FUNCTION.SQL in the db migration directory.
+   *
+   * @param entityManager {@link EntityManager} used to query database.
+   * @param beneficiaryId used to identify the Beneficiary to check claims for.
+   * @return int bitmask denoting which claims have data.
+   */
+  public static int availableClaimsData(EntityManager entityManager, long beneficiaryId) {
+    List<Object> values =
+        entityManager
+            .createNativeQuery(CHECK_CLAIMS_FOR_DATA_SQL)
+            .setParameter("beneIdValue", beneficiaryId)
+            .getResultList();
+
+    return (int) (values != null && values.size() > 0 ? values.get(0) : 0);
```

Comment:
so I tried your suggestion and ITs started failing.
`
[ERROR] gov.cms.bfd.server.war.r4.providers.R4ExplanationOfBenefitResourceProviderTest.testFindByPatientSupportsWildcardClaimTypeV2 -- Time elapsed: 0.013 s <<< ERROR!
java.lang.NullPointerException: Cannot invoke "javax.persistence.Query.setParameter(String, Object)" because the return value of "javax.persistence.EntityManager.createNativeQuery(String, java.lang.Class)" is null
	at gov.cms.bfd.server.war.commons.CommonQueries.availableClaimsData(CommonQueries.java:71)
	at gov.cms.bfd.server.war.r4.providers.R4ExplanationOfBenefitResourceProvider.findByPatient(R4ExplanationOfBenefitResourceProvider.java:375)
	at gov.cms.bfd.server.war.r4.providers.R4ExplanationOfBenefitResourceProviderTest.testFindByPatientSupportsWildcardClaimTypeV2(R4ExplanationOfBenefitResourceProviderTest.java:511)
`
Suspect that the JDBC driver + Hibernate does not have a generic return type of List<Integer> but does support List<Object>. Now one can register new return types, but that is a pretty big chunk of code to gain 1-2 less lines of code.